pkg/email/issue_lcid.go

```diff
@@ -37,7 +38,7 @@ func (c Client) issueLCIDBody(lcid string, expiresAt *time.Time, scope string, n
 }
 
 // SendIssueLCIDEmail sends an email for issuing an LCID
-func (c Client) SendIssueLCIDEmail(ctx context.Context, recipient string, lcid string, expirationDate *time.Time, scope string, nextSteps string, feedback string) error {
+func (c Client) SendIssueLCIDEmail(ctx context.Context, recipient models.EmailAddress, lcid string, expirationDate *time.Time, scope string, nextSteps string, feedback string) error {
```

Comment:
I don't see a change in the function call, just in the type definition here. Is that because we're setting the type in the response from the call to the Fetch User Info client? Just for my peace of mind, did you try recreating the buggy code to see if this catches the issue we saw?

Reply:
@sarboc This change moves the check to be at compile-time, or really at software-engineer-is-coding-time...

The previous bug looked approximately like: 
```go
sendReviewEmail(
			ctx,
			requesterInfo.Email,
			action.Feedback.String,
			intake.ID,
		)
```

... but with the type being the enforcement, to use that function with parameters in the wrong order it would have to look like this:
```go
sendReviewEmail(
			ctx,
			string(requesterInfo.Email),  // requesterInfo.Email is of type model.EmailAddress so you have to un-cast it to use it incorrectly as a string
			EmailAddress(action.Feedback.String), // have to cast Feedback as an EmailAddress, which HOPEFULLY looks/feels very wrong to the programmer at typing time
			intake.ID,
		)
```

At runtime, this does NOT do a check to ensure the contents of `requesterInfo.Email` look like a valid email address... we're (usually) getting that data from the CEDAR API (so we assume they've done their own due diligence). 
But by relying on the type system, this means we will only ever try to plug an `EmailAddress` type in somewhere that requires an `EmailAddress` parameter.

Reply:
@sarboc I tried reversing the order of the arguments like they were to cause that bug and the program will no longer compile. So I think this does prevent that from happening again.

Reply:
Thanks for checking, y'all! Glad to hear this catches the error.